File: api-chi/cmd/routes/movie_category.go
package routes

import (
	"api-chi/cmd/controllers"

	"github.com/go-chi/chi/v5"
)

func MovieCategoryRoutes(r chi.Router) {
	controller := controllers.MovieCategoryController{}

	r.Route("/movie-category", func(r chi.Router) {
		r.Get("/count", controller.Count)
		r.Get("/", controller.GetAll)
		r.Post("/", controller.Create)
		r.Patch("/", controller.Update)
		r.Delete("/{id}", controller.Remove)
	})
}

File: api-fastify/src/api/routes/movie_category.ts
import { FastifyInstance } from "fastify"
import MovieCategoryController from "@/api/controllers/movie_category"

export default async function MovieCategoryRoutes(fastify: FastifyInstance) {
  const controller: MovieCategoryController = new MovieCategoryController()

  fastify.post("/movie-categories", controller.Create.bind(controller))
  fastify.patch("/movie-categories", controller.Update.bind(controller))
  fastify.delete("/movie-categories/:id", controller.Remove.bind(controller))
  fastify.get("/movie-categories/count", controller.Count.bind(controller))
  fastify.get("/movie-categories", controller.GetAll.bind(controller))
}
